pkg/builtin/multi/ceil.go

```diff
@@ -0,0 +1,368 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multi
+
+import (
+	"errors"
+	"fmt"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/ceil"
+
+	"github.com/matrixorigin/matrixone/pkg/builtin"
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/encoding"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend"
+	"github.com/matrixorigin/matrixone/pkg/sql/colexec/extend/overload"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+func init() {
+	extend.FunctionRegistry["ceil"] = builtin.Ceil
+	for _, item := range argsAndRets {
+		overload.AppendFunctionRets(builtin.Ceil, item.args, item.ret)
+	}
+	extend.MultiReturnTypes[builtin.Ceil] = func(es []extend.Extend) types.T {
+		return getMultiReturnType(builtin.Ceil, es)
+	}
+
+	extend.MultiStrings[builtin.Ceil] = func(es []extend.Extend) string {
+		if len(es) > 1 {
+			return fmt.Sprintf("ceil(%s, %s)", es[0], es[1])
+		} else {
+			return fmt.Sprintf("ceil(%s)", es[0])
+		}
+	}
+	overload.OpTypes[builtin.Ceil] = overload.Multi
+	overload.MultiOps[builtin.Ceil] = []*overload.MultiOp{
+		{
+			Min:        1,
+			Max:        2,
+			Typ:        types.T_uint8,
+			ReturnType: types.T_uint8,
+			Fn: func(vecs []*vector.Vector, proc *process.Process, cs []bool) (*vector.Vector, error) {
+				digits := int64(0)
+				vs := vecs[0].Col.([]uint8)
+				if len(vecs) > 1 {
+					if !cs[1] || vecs[1].Typ.Oid != types.T_uint8 {
```

Comment:
the constant integer passed in from the frontend will be casted to a int64, so the vector[1] should be a constant int64 vector

Reply:
ok,i get your point